phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PagingRegionScanner.java

```diff
@@ -61,6 +63,28 @@ public PagingRegionScanner(Region region, RegionScanner scanner, Scan scan) {
 
     private boolean next(List<Cell> results, boolean raw) throws IOException {
         try {
+            byte[] adjustedStartRowKey =
```

Comment:
why do we need to do this check on every next call ? Can't we do this in the constructor ?

Reply:
GROUP BY and ORDER BY on non-pk columns would need to reset this but they can figure out whether to reset the scanner only as part of `scanner#next`. This is why we can't move it to constructor.